src/clj/rems/form_validation.clj

```diff
@@ -27,19 +33,41 @@
   (and (= (:field/type field) :option)
        (not (option-value-valid? field))))
 
+(defn- invalid-column-value? [field]
+  (when (= (:field/type field) :table)
+    (let [columns (set (map :key (:field/columns field)))
+          row-ok? (fn [row] (every? columns (map :column row)))
+          value (:field/value field)]
+      ;; Schema validation guarantees that it's either a s/Str or
+      ;; a [[{:column s/Str :value s/Str}]] so we don't need to check
+      ;; the shape of the data here. However, the default value
+      ;; for :field/value is "", which we do need to tolerate.
+      (if (string? value)
+        (not (str/blank? value))
+        (not (every? row-ok? (:field/value field)))))))
+
 ;; TODO: validate that attachments are actually valid?
 (defn- invalid-attachment-value? [field]
   (and (= (:field/type field) :attachment)
        (not (every? number? (form/parse-attachment-ids (:field/value field))))))
 
+(defn- string-required? [field]
+  (and (not= :table (:field/type field))
+       (not (string? (:field/value field)))
+       (not (nil? (:field/value field)))))
+
 (defn- validate-field-content [field]
   (cond
+    (string-required? field) {:field-id (:field/id field)
+                              :type :t.form.validation/invalid-value} ; TODO better error?
```

Comment:
It applies to all non-table fields, and checks that the type of the answer is string: thus the name. I guess it could also be named something like `wrong-answer-type?`.

Reply:
I think there is something wrong in that name too. I mean the next lines check email, too-long etc. details. This should be something similar, concrete and simple? The error is called `invalid-value` so maybe that or `invalid-string-value`?

Reply:
So it's more like `invalid-non-string-value?` Would that be fine?

I can also follow up with a PR that cleans up rems.form-validation, I think it's grown into a pile of ifs that could use some structuring. Luckily the tests are good.